### PR TITLE
feat(helm): make nginx resolver configurable

### PR DESCRIPTION
**Pull request by thschue**
**What this PR does / why we need it**:
* Currently, in the helm chart it is assumed, that the hostname of the nginx-resolver is set in a format as: {dnsService}.{dnsNamespace}.{clusterDomain}, which is working most of the time.
* In some cases, the cluster domain is unique (e.g. kube-dns.kube-system.svc.905fffc5-ae9a-46c2-8983-d591456b4841.cluster.local.), which makes this a bit challenging to template as it is not so easy to find out the cluster identifier when this is not known (e.g. before the cluster is started).
* Nginx will not come up when it cannot resolve the resolver
* This PR allows specifying a resolver, therefore it is possible to use "kube-dns.kube-system"
* The behavior is almost the same as implemented in mimir: https://github.com/grafana/mimir/pull/2196 

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [X] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [X] Documentation added - Added a comment in the Helm Chart
- [ ] Tests updated
- [x] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: production/helm/loki/values.yaml [hunk 1]
```diff
@@ -1197,6 +1197,8 @@ gateway:
     # -- Allows appending custom configuration to the http block, passed through the `tpl` function to allow templating
     httpSnippet: >-
       {{ if .Values.loki.tenants }}proxy_set_header X-Scope-OrgID $remote_user;{{ end }}
+    # -- Allows to set a custom resolver
```
Comment (JStickler) [inline production/helm/loki/values.yaml:1200]:
```suggestion
    # -- Lets you set a custom resolver
```

File: docs/sources/installation/helm/reference.md [hunk 2]
```diff
@@ -1105,6 +1105,15 @@ See values.yaml
 			<td><pre lang="json">
 "main '$remote_addr - $remote_user [$time_local]  $status '\n        '\"$request\" $body_bytes_sent \"$http_referer\" '\n        '\"$http_user_agent\" \"$http_x_forwarded_for\"';"
 </pre>
+</td>
+		</tr>
+		<tr>
+			<td>gateway.nginxConfig.resolver</td>
+			<td>string</td>
+			<td>Allows to set a custom resolver</td>
```
Comment (JStickler) [inline docs/sources/installation/helm/reference.md:1113]:
```suggestion
			<td>Lets you set a custom resolver</td>
```

**Review by JStickler — APPROVED** [review]
[docs team]  two small rewording suggestions, otherwise LGTM.

**Review by jeschkies — APPROVED** [review]
Awesome! Could you also bump the patch version of the Helm chart?

File: CHANGELOG.md [hunk 3]
```diff
@@ -31,8 +31,10 @@
 * [9184](https://github.com/grafana/loki/pull/9184) **periklis**: Bump dskit to introduce IPv6 support for memberlist
 * [9357](https://github.com/grafana/loki/pull/9357) **Indransh**: Add HTTP API to change the log level at runtime
 * [9431](https://github.com/grafana/loki/pull/9431) **dannykopping**: Add more buckets to `loki_memcache_request_duration_seconds` metric; latencies can increase if using memcached with NVMe
+* [9491](https://github.com/grafana/loki/pull/9491) **thschue**: Helm: Make nginx resolver dns name or ip address configurable
 * [8684](https://github.com/grafana/loki/pull/8684) **oleksii-boiko-ua**: Helm: Add hpa templates for read, write and backend components.
 * [9604](https://github.com/grafana/loki/pull/9604) **dannykopping**: Querier: configurable writeback queue bytes size
+* [9491](https://github.com/grafana/loki/pull/9491) **thschue**: Helm: Make nginx resolver dns name or ip address configurable
```
Comment (MichelHollands) [inline CHANGELOG.md:37]: These are duplicate entries

**Pull request closed without merging**